### PR TITLE
Migrate to AndroidX. Fixes #490

### LINKED ISSUE #490: Migrate to AndroidX
Please migrate Conductor to the new [AndroidX](https://developer.android.com/jetpack/androidx/) support libraries which are stable as of 21st September.

Thank you!

### PR DESCRIPTION
**Pull request by PaulWoitaschek**

### CODE REVIEW

**Review by ShaishavGandhi — APPROVED** [review]

File: demo/src/main/java/com/bluelinelabs/conductor/demo/controllers/TransitionDemoController.java [hunk 1]
```diff
@@ -56,9 +56,12 @@ public static TransitionDemo fromIndex(int index) {
...
 
-    @BindView(R.id.tv_title) TextView tvTitle;
-    @BindView(R.id.btn_next) FloatingActionButton btnNext;
-    @BindView(R.id.transition_root) View containerView;
+    @BindView(R.id.tv_title)
+    TextView tvTitle;
+    @BindView(R.id.btn_next)
+    FloatingActionButton btnNext;
+    @BindView(R.id.transition_root)
+    View containerView;
```
Comment (EricKuck) [inline demo/src/main/java/com/bluelinelabs/conductor/demo/controllers/TransitionDemoController.java]: Could we reformat these so the annotation and field are on the same lines again?

**Comment by EricKuck** [comment]
Thanks so much for tackling this! Everything looks great!

File: demo/src/main/java/com/bluelinelabs/conductor/demo/controllers/TransitionDemoController.java (hunk 1 above)
Comment (PaulWoitaschek) [reply demo/src/main/java/com/bluelinelabs/conductor/demo/controllers/TransitionDemoController.java]: Thanks, done.

**Comment by PaulWoitaschek** [comment]
@EricKuck Do you prefere a squash over rebase?

**Comment by EricKuck** [comment]
I generally do squash. No strong preference really.

**Comment by mradzinski** [comment]
@PaulWoitaschek @EricKuck Are we to expect a release soon? If so maybe version it 3.0.0-rc1? We have 100% coverage on a big project which I just finished migrating, so I can see of running the tests, have the QA team do their thing and report back in a day or two.

**Comment by dcostinett** [comment]
Hi @EricKuck release planned soon? I'd really love to use the AndroidX version.

**Comment by VillePakarinen** [comment]
@EricKuck Same question as above. Can we expect a release anytime soon?

**Comment by EricKuck** [comment]
Sorry for the delay everyone. A 3.0.0-rc1 build is getting pushed today!

Edit: It's been pushed and is now processing with maven central. Should be available very shortly.

**Comment by dimsuz** [comment]
How does `3.0.0-rc1` work for everybody? Maybe it's time to do `3.0.0`? :)

**Comment by mradzinski** [comment]
Working fine here @dimsuz. Have had it in prod for at least 5 months already.

**Pull request merged**